### DISCUSSION TITLE
Get Button Click When Message Created with DiscordInteractionResponseBuilder

### DISCUSSION THREAD

**Discussion by alex182** [DSharpPlus/DSharpPlus#1453] [Q&A] [answered]
I am sending in an ephemeral interaction message using the DiscordInteractionResponseBuilder.  The message has a button and I would like to be able to take action when it is clicked, how would I do that if `interactionContext.CreateResponseAsync` doesnt return a message? 

``` 
        var gifResponse = new DiscordInteractionResponseBuilder()
               .AddComponents(button1)
               .AddEmbed(embed)
               .AsEphemeral(true);

            var interactivity = interactionContext.Client.GetInteractivity();
            await interactionContext.CreateResponseAsync(gifResponse);

**Comment by ploring04** [comment, marked as answer]
Pretty sure you need to make sure you are handling the ComponentInteractionCreated event from the DiscordClient.  That should give you the Id of the button along with the DiscordMessage among other fields in the ComponentInteractionCreateEventArgs.  It also has the Interaction object that can be used.

**Reply by alex182** [reply]
that did it, thank you!
